### PR TITLE
meson: New version 0.62.2

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by eschnett** [comment]
@spackbot hello

**Comment by eschnett** [comment]
@spackbot help

**Comment by eschnett** [comment]
@spackbot run pipeline

**Comment by michaelkuhn** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/meson/package.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Meson(PythonPackage):
 
     maintainers = ['michaelkuhn']
 
+    version('0.62.2', sha256='a7669e4c4110b06b743d57cc5d6432591a6677ef2402139fe4f3d42ac13380b0')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/meson/package.py:21]: The checksum seems to be different for me. Yours seems to be from the actual release tarball, while the package downloads the autogenerated one. We can change to the proper tarballs but then you probably need to add a `url_for_version` function.
```
$ spack checksum meson 0.62.2       
==> Found 1 version of meson:
  
  0.62.2  https://github.com/mesonbuild/meson/archive/0.62.2.tar.gz

==> Fetching https://github.com/mesonbuild/meson/archive/0.62.2.tar.gz

    version('0.62.2', sha256='97108f4d9bb16bc758c44749bd25ec7d42c6a762961efbed8b7589a2a3551ea6')
```

**Review by michaelkuhn — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/meson/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/meson/package.py:21]: I followed Spack's guidance to add the new version (`spack checksum`). Could this lead to such an inconsistency? If I recall correctly, 

This package shouldn't need anything special. What is your recommendation here? What causes this problem? If I wanted to use a different checksum, how should I calculate it? If I wanted to add an `url_for_version`, what url should that be?

**Comment by eschnett** [comment]
Superseded by https://github.com/spack/spack/pull/31005

**Pull request closed without merging**